core/src/main/scala/com/snowplowanalytics/snowplow/event/recovery/config/conditions/Condition.scala

```diff
@@ -27,21 +26,5 @@ case class Condition(
   path: String,
   value: Matcher
 ) {
-  def check(str: Json): Boolean =
-    nested(str, path).map(value.checks).getOrElse(false)
-
-  /**
-    * Checks fields that are at a lower JSON level
-    * ie. $.payload.parameters.aid
-    */
-  private[this] def nested(obj: Json, path: String): Option[Json] = {
-    @tailrec
-    def apply(root: ACursor, nodes: Seq[String]): Option[Json] = nodes match {
-      case Seq() =>
-        root.focus
-      case Seq(h, t @ _*) =>
-        apply(root.downField(h), t)
-    }
-    apply(obj.hcursor, json.path(path))
-  }
+  def check(str: Json): Boolean = inspect.check(value)(json.path(path))(str).flatMap(_.as[Boolean]).getOrElse(false)
```

Comment:
Is the JSON on the right of the `Recovering` guaranteed to be castable to Boolean?

Reply:
If successful it returns a castable one, if it is not castable then comparison is surely false and we don't really care why.